numba/cpython/setobj.py

```diff
@@ -103,6 +103,11 @@ def is_hash_used(context, builder, h):
     return builder.icmp_unsigned('<', h, deleted)
 
 
+def all_set(*args):
+    return all([isinstance(typ, types.Set) for typ in args]) and \
+        all([args[0].dtype == s.dtype for s in args])
```

Comment:
I just ran the unit tests for this and found that whilst a mixed dtype operation is correctly rejected, the error message does not describe the problem very well, it just reports `No implementation of function...<some function>... found for signature...` with no indication as to the actual problem. Maybe this function should:
* raise something like `TypingError(f"All arguments must be Sets, got {args}")` if the types are not all `Set`s.
* raise  something like `TypingError(f"All Sets must be of the same type, got {args}")` if the dtypes are not all the same.

What do you think?

Reply:
Thanks for adding the differentiating error messages in 5510cb6, any chance you could also please add test(s) to demonstrate that each path is correctly taken, i.e. check the correct exceptions are raised for the known erroneous input? Many thanks.

Reply:
Done!